player/command.c

```diff
@@ -2837,10 +2838,12 @@ static int mp_property_ambient_light(void *ctx, struct m_property *prop,
                                      int action, void *arg)
 {
     MPContext *mpctx = ctx;
-    struct vo *vo = mpctx->video_out;
+    if (!mpctx->als_state) {
+        mpctx->als_state = mp_als_create(mpctx, mpctx);
+    }
+
     double lux;
-    if (!vo || vo_control(vo, VOCTRL_GET_AMBIENT_LUX, &lux) < 1)
-        return M_PROPERTY_UNAVAILABLE;
+    mp_als_get_lux(mpctx->als_state, &lux);
```

Comment:
needs to check the return value since negative values are failures. otherwise it always tries to use `lux`, even if the pointer wasn't set to anything within `mp_als_get_lux`.